rpc/handlers.go

```diff
@@ -308,19 +317,44 @@ func (h *Handler) BlockWithTxs(id BlockID) (*BlockWithTxs, *jsonrpc.Error) {
 		txs[index] = AdaptTransaction(txn)
 	}
 
-	l1H, jsonErr := h.l1Head()
-	if jsonErr != nil {
-		return nil, jsonErr
+	status, rpcErr := h.blockStatus(id, block)
+	if rpcErr != nil {
+		return nil, rpcErr
 	}
 
-	status := BlockAcceptedL2
-	if id.Pending {
-		status = BlockPending
-	} else if isL1Verified(block.Number, l1H) {
-		status = BlockAcceptedL1
+	return &BlockWithTxs{
+		Status:       status,
+		BlockHeader:  adaptBlockHeader(block.Header),
+		Transactions: txs,
+	}, nil
+}
+
+func (h *Handler) BlockWithReceipts(id BlockID) (*BlockWithReceipts, *jsonrpc.Error) {
+	block, rpcErr := h.blockByID(&id)
+	if rpcErr != nil {
+		return nil, rpcErr
 	}
 
-	return &BlockWithTxs{
+	txs := make([]TransactionWithReceipt, len(block.Transactions))
+	for index, txn := range block.Transactions {
+		r := block.Receipts[index]
+		receipt, rpcErr := h.transactionReceipt(r, txn, block.Hash, block.Number)
+		if rpcErr != nil {
```

Comment:
you can pass `nil` as block hash here. ``lock_hash` an` `block_number` fields are optional in `BlockWithReceipt`